### PR TITLE
Correct text selection pivot points

### LINKED ISSUE #57679: Add awareness for cursor position in selection-related behaviors across desktop platforms
My recollection from working on Gecko macOS text selection many years ago is that desktop platforms have different expected behaviors around how keyboard text selection interacts with existing selections, especially mouse-based. In particular: if you select some text, then shift-left-arrow or shift-right-arrow, what happens? Does it matter if the text was selected from left to right or right to left with the mouse?

We should audit this behavior on each desktop platform in a native text field, then make sure the framework handling matches on each platform.

### PR DESCRIPTION
**Pull request by justinmc**
## Description

The pivot point is the selection edge that does not move:

| Pivot point left | Pivot point right |
| -------------- | ---------------- |
| pivot_left | pivot_right |

Currently, Flutter always puts the pivot point on the right, but as you can see in the gifs above of native OSX, other platforms put it where the selection began.  This PR updates Flutter to do the same.

There are more complicated platform-specific behaviors in other selection scenarios, such as when using word and line modifier keys, but that isn't handled by this PR.  See https://github.com/flutter/flutter/issues/57679#issuecomment-742848741.

## Related Issues

Closes https://github.com/flutter/flutter/issues/57679

## Tests

I added the following tests:

  * Select some text, hold shift, and move with the arrow keys. Expect the correct behavior.
  * Also, I updated a couple of tests that were expecting reversed base/extent.

### CODE REVIEW

**Comment by justinmc** [comment]
Reviewers: Can you think of any reason why we were swapping base/extent before?  This seems to work better without the swap and doesn't seem to break major tests, but I'm keeping an eye out.

File: packages/flutter/lib/src/rendering/editable.dart [hunk 1]
```diff
@@ -1849,12 +1854,8 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
       ? null
       : _textPainter.getPositionForOffset(globalToLocal(to - _paintOffset));
 
-    int baseOffset = fromPosition.offset;
-    int extentOffset = fromPosition.offset;
-    if (toPosition != null) {
-      baseOffset = math.min(fromPosition.offset, toPosition.offset);
-      extentOffset = math.max(fromPosition.offset, toPosition.offset);
-    }
+    final int baseOffset = fromPosition.offset;
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/rendering/editable.dart:1857]: This seems to revert the changes made in https://github.com/flutter/flutter/pull/29395. According to @mdebbar 

> The problem is if you drag from left to right, the text will be selected as expected. But if you drag from right to left, then baseOffset will be greater than extentOffset and it'll be collapsed down to a cursor. So dragging from right to left is broken, and this PR fixes it.

Not sure why the tests are passing. But it sounds like we're not handling baseOffset > extentOffset correctly?

Comment (justinmc) [reply packages/flutter/lib/src/rendering/editable.dart:1857]: Ah thanks for tracking that PR down.  It seems like the bug that that fixed, where the cursor would collapsed when selecting right to left, no longer happens.  Everything still seems to be fine with this PR here by my testing.

I should point out that we're still reversing base and extent in the underlying TextRange.  That seems to have been around for a long time, though.

https://github.com/flutter/flutter/blob/72aa23eb935dc9d1d667163e7ffffcb1e7319de8/packages/flutter/lib/src/services/text_editing.dart#L18-L26

**Review by LongCatIsLooong — APPROVED** [review]
